### PR TITLE
insights: improve update insight flow 

### PR DESCRIPTION
**Pull request by leonore**
part of #42116

previously, in the update insight mutation for existing insight series we 
1. remove the existing series from the view (this soft-deletes the series)
2. create a new series and attach it to the view (this does a new attach series to view step)
3. update the insight view to be attached to the old series

call 3 was completely redundant as the series was no longer in use. it would just affect 0 rows each time.

now, we check whether the series existed before. if it didn't we just create a new one. otherwise we check what has changed. if it's only presentation metadata we won't trigger an insight recalculation.

## Test plan

Added an integration test and manually validated that before/after did not change when creating some insights

see it in action: https://www.loom.com/share/2b893d3691cd40df81a910e60bd42ab5

with `sg start enterprise-codeinsights` running:
```bash
go test -long -run TestUpdateInsight -base-url "https://sourcegraph.test:3443" -v

2022/11/23 17:29:56 Site admin authenticated: gqltest-admin
=== RUN   TestUpdateInsight
--- PASS: TestUpdateInsight (1.37s)
PASS
ok  	github.com/sourcegraph/sourcegraph/dev/gqltest	2.014s
```

### CODE REVIEW

File: enterprise/internal/insights/resolvers/insight_view_resolvers.go [hunk 1]
```diff
@@ -590,37 +590,76 @@ func (r *Resolver) UpdateLineChartSearchInsight(ctx context.Context, args *graph
...
+					if err != nil {
+						return nil, errors.Wrap(err, "RemoveViewSeries")
+					}
+					_, err = createAndAttachSeries(ctx, tx, seriesFillStrategy, view, series)
+					if err != nil {
+						return nil, errors.Wrap(err, "createAndAttachSeries")
+					}
+				} else {
+					// Otherwise we simply update the series' presentation metadata.
+					err = tx.UpdateViewSeries(ctx, *series.SeriesId, view.ID, types.InsightViewSeriesMetadata{
```
Comment (chwarwick) [inline enterprise/internal/insights/resolvers/insight_view_resolvers.go:629]: Aren't there more series properties that can be updated?  The series `Name` or `Label`  not sure which it's called comes to mind.

File: enterprise/internal/insights/resolvers/insight_view_resolvers.go [hunk 2]
```diff
@@ -590,37 +590,76 @@ func (r *Resolver) UpdateLineChartSearchInsight(ctx context.Context, args *graph
...
+					return nil, errors.Wrap(err, "createAndAttachSeries")
+				}
 			}
 		}
+
 	}
 	return &insightPayloadResolver{baseInsightResolver: r.baseInsightResolver, validator: permissionsValidator, viewId: insightViewId}, nil
 }
 
+func existingSeriesHasChanged(new graphqlbackend.LineChartSearchInsightDataSeriesInput, existing types.InsightViewSeries) bool {
```
[the remark this answers is not in the document]
Comment (leonore) [reply enterprise/internal/insights/resolvers/insight_view_resolvers.go:650]: good point, thanks

File: enterprise/internal/insights/resolvers/insight_view_resolvers.go (hunk 1 above)
Comment (leonore) [reply enterprise/internal/insights/resolvers/insight_view_resolvers.go:629]: yes, that's what this case covers. if the only change is the label or the stroke colour then that's all we update. it's not stored on the `insight_series` object

Comment (chwarwick) [reply enterprise/internal/insights/resolvers/insight_view_resolvers.go:629]: 🤦‍♂️

**Review by chwarwick — APPROVED** [review]

**Pull request merged**